src/api/procedures/addInstruction.ts

```diff
@@ -98,128 +118,198 @@ export async function prepareAddInstruction(
     });
   }
 
-  if (!legs.length) {
+  const addAndAffirmInstructions: [
+    u64,
+    SettlementType,
+    Moment | null,
+    Moment | null,
+    {
+      from: PortfolioId;
+      to: PortfolioId;
+      asset: Ticker;
+      amount: Balance;
+    }[],
+    PortfolioId[]
+  ][] = [];
+  const addInstructions: [
+    u64,
+    SettlementType,
+    Moment | null,
+    Moment | null,
+    {
+      from: PortfolioId;
+      to: PortfolioId;
+      asset: Ticker;
+      amount: Balance;
+    }[]
+  ][] = [];
+
+  const noLegs: number[] = [];
+  const noFutureBlock: number[] = [];
+  const noFutureDate: number[] = [];
```

Comment:
legErrIndexes
endBlockErrIndexes
datesErrIndexes

Also, they should count from 0 up, no need to mess up with the indexes